{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import argparse\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathPrefix = '/Users/friedman/Desktop/mnt'\n",
    "sys.path.append(pathPrefix + '/ifs/work/taylorlab/friedman/myUtils')\n",
    "import analysis_utils\n",
    "import mutationSigUtils\n",
    "import maf_analysis_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#directory where configuration type files are saved\n",
    "configFileDir = pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'maf_analysis_utils' from '/Users/friedman/Desktop/mnt/ifs/work/taylorlab/friedman/myUtils/maf_analysis_utils.py'>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reload(maf_analysis_utils)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**i. MAKE Indel fraction files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#i\n",
    "#MAKE Indel fraction files\n",
    "sigs = pd.read_table(pathPrefix + '/ifs/res/taylorlab/impact_sigs/mixedpact_data_mutations_unfiltered.sigs.tab.txt')\n",
    "sigs['dominantSignature'] = sigs.apply(lambda row: mutationSigUtils.get_dominant_signature(row.to_dict(), cols=None, prefix='mean', notEnoughMuts= True), axis=1)\n",
    "\n",
    "impactFilteredMaf = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/myAdjustedDataFiles/tempScriptFiles/filteredMafAnnotated_trinuc.maf')\n",
    "\n",
    "minCasesToDeriveIndelFracEstimate = 25\n",
    "listOfDicts = []\n",
    "for sig in set(sigs['dominantSignature']):\n",
    "    if sig != 'insufficientMutBurden':\n",
    "        dominantSigCases = set(sigs[(sigs['dominantSignature'] == sig) &\n",
    "                                    (sigs['Nmut_Mb'] > 10) #only take cases with a non-trivial mutation burden\n",
    "                                    &(sigs[sig] > 1.0/3)] #only take cases where the dominant sig is greater than 33% of total signature burden\n",
    "    ['Tumor_Sample_Barcode'])\n",
    "        \n",
    "        if len(dominantSigCases) > minCasesToDeriveIndelFracEstimate:\n",
    "            dominantSigMaf = impactFilteredMaf[impactFilteredMaf['Tumor_Sample_Barcode'].isin(dominantSigCases)]\n",
    "            listOfDicts.append({'Signature': re.sub('mean_', 'Signature.', sig), \n",
    "                                'Indel_Fraction': maf_analysis_utils.get_indel_frac_for_cohort(dominantSigMaf)})\n",
    "    \n",
    "panImpactIndelFrac = maf_analysis_utils.get_indel_frac_for_cohort(impactFilteredMaf)\n",
    "listOfDicts.append({'Signature': 'All_IMPACT_Cases', 'Indel_Fraction': panImpactIndelFrac})\n",
    "df = pd.DataFrame(listOfDicts)\n",
    "df.to_csv(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/indelRateBySignature.tsv', index=False, sep='\\t')      "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**ii. summarize msi area for indels**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "msiIndelInfo = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/IMPACT_repeat_stats.txt')\n",
    "msiIndelInfo['msiBp'] = msiIndelInfo.apply(lambda row: row['bp']*row['rep.pct'], axis=1)\n",
    "msiIndelInfo['msiProb'] = msiIndelInfo['msiBp'].apply(lambda x: 1.0*x/np.nansum(msiIndelInfo['bp']))\n",
    "    \n",
    "msiProbByGene = msiIndelInfo[['Hugo_Symbol', 'bp', 'msiProb']]\n",
    "msiProbByGene.to_csv(os.path.join(configFileDir, 'relativeMsiProbabilityByGene.tsv'), index = False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**iii. Make per gene mut files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "impactGenes = set(['ABL1', 'ACVR1', 'AGO2', 'AKT1', 'AKT2', 'AKT3', 'ALK', 'ALOX12B', 'ANKRD11', 'APC', 'AR', 'ARAF', 'ARID1A', 'ARID1B', 'ARID2', 'ARID5B', 'ASXL1', 'ASXL2', 'ATM', 'ATR', 'ATRX', 'AURKA', 'AURKB', 'AXIN1', 'AXIN2', 'AXL', 'B2M', 'BABAM1', 'BAP1', 'BARD1', 'BBC3', 'BCL10', 'BCL2', 'BCL2L1', 'BCL2L11', 'BCL6', 'BCOR', 'BIRC3', 'BLM', 'BMPR1A', 'BRAF', 'BRCA1', 'BRCA2', 'BRD4', 'BRIP1', 'BTK', 'CALR', 'CARD11', 'CARM1', 'CASP8', 'CBFB', 'CBL', 'CCND1', 'CCND2', 'CCND3', 'CCNE1', 'CD274', 'CD276', 'CD79A', 'CD79B', 'CDC42', 'CDC73', 'CDH1', 'CDK12', 'CDK4', 'CDK6', 'CDK8', 'CDKN1A', 'CDKN1B', 'CDKN2A', 'CDKN2B', 'CDKN2C', 'CEBPA', 'CENPA', 'CHEK1', 'CHEK2', 'CIC', 'CREBBP', 'CRKL', 'CRLF2', 'CSDE1', 'CSF1R', 'CSF3R', 'CTCF', 'CTLA4', 'CTNNB1', 'CUL3', 'CXCR4', 'CYLD', 'CYSLTR2', 'DAXX', 'DCUN1D1', 'DDR2', 'DICER1', 'DIS3', 'DNAJB1', 'DNMT1', 'DNMT3A', 'DNMT3B', 'DOT1L', 'DROSHA', 'DUSP4', 'E2F3', 'EED', 'EGFL7', 'EGFR', 'EIF1AX', 'EIF4A2', 'EIF4E', 'ELF3', 'EP300', 'EPAS1', 'EPCAM', 'EPHA3', 'EPHA5', 'EPHA7', 'EPHB1', 'ERBB2', 'ERBB3', 'ERBB4', 'ERCC2', 'ERCC3', 'ERCC4', 'ERCC5', 'ERF', 'ERG', 'ERRFI1', 'ESR1', 'ETV1', 'ETV6', 'EZH1', 'EZH2', 'FAM123B', 'FAM175A', 'FAM46C', 'FAM58A', 'FANCA', 'FANCC', 'FAT1', 'FBXW7', 'FGF19', 'FGF3', 'FGF4', 'FGFR1', 'FGFR2', 'FGFR3', 'FGFR4', 'FH', 'FLCN', 'FLT1', 'FLT3', 'FLT4', 'FOXA1', 'FOXL2', 'FOXO1', 'FOXP1', 'FUBP1', 'FYN', 'GATA1', 'GATA2', 'GATA3', 'GLI1', 'GNA11', 'GNAQ', 'GNAS', 'GPS2', 'GREM1', 'GRIN2A', 'GSK3B', 'H3F3A', 'H3F3B', 'H3F3C', 'HGF', 'HIST1H1C', 'HIST1H2BD', 'HIST1H3A', 'HIST1H3B', 'HIST1H3C', 'HIST1H3D', 'HIST1H3E', 'HIST1H3F', 'HIST1H3G', 'HIST1H3H', 'HIST1H3I', 'HIST1H3J', 'HIST2H3C', 'HIST2H3D', 'HIST3H3', 'HLA-A', 'HLA-B', 'HNF1A', 'HOXB13', 'HRAS', 'ICOSLG', 'ID3', 'IDH1', 'IDH2', 'IFNGR1', 'IGF1', 'IGF1R', 'IGF2', 'IKBKE', 'IKZF1', 'IL10', 'IL7R', 'INHA', 'INHBA', 'INPP4A', 'INPP4B', 'INPPL1', 'INSR', 'IRF4', 'IRS1', 'IRS2', 'JAK1', 'JAK2', 'JAK3', 'JUN', 'KDM5A', 'KDM5C', 'KDM6A', 'KDR', 'KEAP1', 'KIT', 'KLF4', 'KMT2B', 'KMT5A', 'KNSTRN', 'KRAS', 'LATS1', 'LATS2', 'LMO1', 'LYN', 'MALT1', 'MAP2K1', 'MAP2K2', 'MAP2K4', 'MAP3K1', 'MAP3K13', 'MAP3K14', 'MAPK1', 'MAPK3', 'MAPKAP1', 'MAX', 'MCL1', 'MDC1', 'MDM2', 'MDM4', 'MED12', 'MEF2B', 'MEN1', 'MET', 'MGA', 'MITF', 'MLH1', 'KMT2A', 'KMT2B', 'KMT2C', 'MPL', 'MRE11A', 'MSH2', 'MSH3', 'MSH6', 'MSI1', 'MSI2', 'MST1', 'MST1R', 'MTOR', 'MUTYH', 'MYC', 'MYCL1', 'MYCN', 'MYD88', 'MYOD1', 'NBN', 'NCOA3', 'NCOR1', 'NEGR1', 'NF1', 'NF2', 'NFE2L2', 'NFKBIA', 'NKX2-1', 'NKX3-1', 'NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4', 'NPM1', 'NRAS', 'NSD1', 'NTHL1', 'NTRK1', 'NTRK2', 'NTRK3', 'NUF2', 'NUP93', 'PAK1', 'PAK7', 'PALB2', 'PARK2', 'PARP1', 'PAX5', 'PBRM1', 'PDCD1', 'PDCD1LG2', 'PDGFRA', 'PDGFRB', 'PDPK1', 'PGR', 'PHOX2B', 'PIK3C2G', 'PIK3C3', 'PIK3CA', 'PIK3CB', 'PIK3CD', 'PIK3CG', 'PIK3R1', 'PIK3R2', 'PIK3R3', 'PIM1', 'PLCG2', 'PLK2', 'PMAIP1', 'PMS1', 'PMS2', 'PNRC1', 'POLD1', 'POLE', 'PPARG', 'PPM1D', 'PPP2R1A', 'PPP4R2', 'PPP6C', 'PRDM1', 'PRDM14', 'PREX2', 'PRKAR1A', 'PRKCI', 'PRKD1', 'PTCH1', 'PTEN', 'PTP4A1', 'PTPN11', 'PTPRD', 'PTPRS', 'PTPRT', 'RAB35', 'RAC1', 'RAC2', 'RAD21', 'RAD50', 'RAD51', 'RAD51C', 'RAD51L1', 'RAD51L3', 'RAD52', 'RAD54L', 'RAF1', 'RARA', 'RASA1', 'RB1', 'RBM10', 'RECQL', 'RECQL4', 'REL', 'RET', 'RFWD2', 'RHEB', 'RHOA', 'RICTOR', 'RIT1', 'RNF43', 'ROS1', 'RPS6KA4', 'RPS6KB2', 'RPTOR', 'RRAGC', 'RRAS', 'RRAS2', 'RTEL1', 'RUNX1', 'RXRA', 'RYBP', 'SDHA', 'SDHAF2', 'SDHB', 'SDHC', 'SDHD', 'SESN1', 'SESN2', 'SESN3', 'SETD2', 'SF3B1', 'SH2B3', 'SH2D1A', 'SHOC2', 'SHQ1', 'SLX4', 'SMAD2', 'SMAD3', 'SMAD4', 'SMARCA4', 'SMARCB1', 'SMARCD1', 'SMO', 'SMYD3', 'SOCS1', 'SOS1', 'SOX17', 'SOX2', 'SOX9', 'SPEN', 'SPOP', 'SPRED1', 'SRC', 'SRSF2', 'STAG2', 'STAT3', 'STAT5A', 'STAT5B', 'STK11', 'STK19', 'STK40', 'SUFU', 'SUZ12', 'SYK', 'TAP1', 'TAP2', 'TBX3', 'TCEB1', 'TCF3', 'TCF7L2', 'TEK', 'TERT', 'TET1', 'TET2', 'TGFBR1', 'TGFBR2', 'TMEM127', 'TMPRSS2', 'TNFAIP3', 'TNFRSF14', 'TOP1', 'TP53', 'TP53BP1', 'TP63', 'TRAF2', 'TRAF7', 'TSC1', 'TSC2', 'TSHR', 'U2AF1', 'UPF1', 'VEGFA', 'VHL', 'VTCN1', 'WHSC1', 'WHSC1L1', 'WT1', 'WWTR1', 'XIAP', 'XPO1', 'XRCC2', 'YAP1', 'YES1', 'ZFHX3', 'ZRSR2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "impactMaf = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/myAdjustedDataFiles/tempScriptFiles/filteredMafAnnotated_trinuc.maf')\n",
    "impactMafGenesOnly = impactMaf.drop_duplicates(subset=['Hugo_Symbol'])\n",
    "chromosomeDict = dict(zip(impactMafGenesOnly['Hugo_Symbol'], impactMafGenesOnly['Chromosome']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "fileDir = pathPrefix + '/ifs/work/taylorlab/friedman/myAdjustedDataFiles/simulatedMafs'\n",
    "writeDir = pathPrefix + '/ifs/work/taylorlab/friedman/myAdjustedDataFiles/simulatedMafs/geneMutMafs'\n",
    "allFilesInOutputDir = set(os.listdir(writeDir))\n",
    "for gene in impactGenes:\n",
    "    if gene in chromosomeDict:\n",
    "        chromosome = chromosomeDict[gene]\n",
    "        chrFilename = 'chr' + chromosome + '_allmut_mafAnno_trinuc.maf'\n",
    "        chrFilepath = os.path.join(fileDir, chrFilename)\n",
    "        outputFileName = gene + '_allmut_mafAnno_trinuc.maf'\n",
    "        if outputFileName in allFilesInOutputDir:\n",
    "            #TODO validate that the genes have the correct number of mutations\n",
    "            print outputFileName, ' already exists, skipping'\n",
    "        else:\n",
    "            outputFilePath = os.path.join(writeDir, outputFileName)\n",
    "            cmd = 'grep -n ' + gene + ' ' + chrFilepath + ' * > ' + outputFilePath\n",
    "            print 'executing ', cmd\n",
    "            os.popen(cmd)\n",
    "print 'done'\n",
    " \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**MAKE NMUT/NMUT_MB based on filtered maf**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "impactFilteredMaf = analysis_utils.load_in_df_with_progress(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/all_impact_mutations_annotated_cohort.maf', 275171)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "nmutDf = maf_analysis_utils.summarize_nmut_info_across_cohort(impactFilteredMaf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Nmut</th>\n",
       "      <th>Nmut_Mb</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>3.935156</td>\n",
       "      <td>P-0006360-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0.877715</td>\n",
       "      <td>P-0029754-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0033656-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>8.777150</td>\n",
       "      <td>P-0022073-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>10.532580</td>\n",
       "      <td>P-0022194-T02-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>5.902735</td>\n",
       "      <td>P-0005012-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>5.266290</td>\n",
       "      <td>P-0031527-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>20</td>\n",
       "      <td>17.554300</td>\n",
       "      <td>P-0030628-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0026122-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>6</td>\n",
       "      <td>5.266290</td>\n",
       "      <td>P-0015432-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3</td>\n",
       "      <td>2.633145</td>\n",
       "      <td>P-0025446-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>9</td>\n",
       "      <td>8.854102</td>\n",
       "      <td>P-0009590-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0032598-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "      <td>1.115244</td>\n",
       "      <td>P-0002703-T01-IM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>4</td>\n",
       "      <td>3.510860</td>\n",
       "      <td>P-0034494-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>18</td>\n",
       "      <td>15.798870</td>\n",
       "      <td>P-0032035-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2</td>\n",
       "      <td>1.755430</td>\n",
       "      <td>P-0020236-T03-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2</td>\n",
       "      <td>1.755430</td>\n",
       "      <td>P-0015903-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>6</td>\n",
       "      <td>5.902735</td>\n",
       "      <td>P-0003514-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>16</td>\n",
       "      <td>14.043440</td>\n",
       "      <td>P-0018316-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>38</td>\n",
       "      <td>33.353170</td>\n",
       "      <td>P-0017068-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>3</td>\n",
       "      <td>2.633145</td>\n",
       "      <td>P-0024645-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>8</td>\n",
       "      <td>7.870313</td>\n",
       "      <td>P-0005271-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>6</td>\n",
       "      <td>5.902735</td>\n",
       "      <td>P-0004762-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>8</td>\n",
       "      <td>7.021720</td>\n",
       "      <td>P-0026370-T02-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>11</td>\n",
       "      <td>9.654865</td>\n",
       "      <td>P-0017116-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0032590-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>12</td>\n",
       "      <td>10.532580</td>\n",
       "      <td>P-0036268-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>21</td>\n",
       "      <td>18.432015</td>\n",
       "      <td>P-0024571-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>6</td>\n",
       "      <td>5.902735</td>\n",
       "      <td>P-0009644-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32028</th>\n",
       "      <td>3</td>\n",
       "      <td>2.633145</td>\n",
       "      <td>P-0036060-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32029</th>\n",
       "      <td>1</td>\n",
       "      <td>1.115244</td>\n",
       "      <td>P-0002203-T01-IM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32030</th>\n",
       "      <td>8</td>\n",
       "      <td>7.021720</td>\n",
       "      <td>P-0021281-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32031</th>\n",
       "      <td>1</td>\n",
       "      <td>0.877715</td>\n",
       "      <td>P-0018310-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32032</th>\n",
       "      <td>1</td>\n",
       "      <td>0.877715</td>\n",
       "      <td>P-0039611-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32033</th>\n",
       "      <td>8</td>\n",
       "      <td>7.021720</td>\n",
       "      <td>P-0022819-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32034</th>\n",
       "      <td>4</td>\n",
       "      <td>3.935156</td>\n",
       "      <td>P-0004807-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32035</th>\n",
       "      <td>5</td>\n",
       "      <td>4.918946</td>\n",
       "      <td>P-0013870-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32036</th>\n",
       "      <td>8</td>\n",
       "      <td>7.021720</td>\n",
       "      <td>P-0005143-T04-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32037</th>\n",
       "      <td>20</td>\n",
       "      <td>17.554300</td>\n",
       "      <td>P-0023469-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32038</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0015602-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32039</th>\n",
       "      <td>4</td>\n",
       "      <td>3.935156</td>\n",
       "      <td>P-0010234-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32040</th>\n",
       "      <td>3</td>\n",
       "      <td>2.633145</td>\n",
       "      <td>P-0023987-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32041</th>\n",
       "      <td>3</td>\n",
       "      <td>3.345731</td>\n",
       "      <td>P-0002254-T01-IM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32042</th>\n",
       "      <td>1</td>\n",
       "      <td>0.983789</td>\n",
       "      <td>P-0009059-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32043</th>\n",
       "      <td>7</td>\n",
       "      <td>6.886524</td>\n",
       "      <td>P-0010267-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32044</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0035066-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32045</th>\n",
       "      <td>2</td>\n",
       "      <td>1.967578</td>\n",
       "      <td>P-0008052-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32046</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0034412-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32047</th>\n",
       "      <td>6</td>\n",
       "      <td>5.266290</td>\n",
       "      <td>P-0030111-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32048</th>\n",
       "      <td>6</td>\n",
       "      <td>6.691462</td>\n",
       "      <td>P-0001376-T01-IM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32049</th>\n",
       "      <td>5</td>\n",
       "      <td>4.918946</td>\n",
       "      <td>P-0011240-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32050</th>\n",
       "      <td>2</td>\n",
       "      <td>1.755430</td>\n",
       "      <td>P-0029163-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32051</th>\n",
       "      <td>2</td>\n",
       "      <td>1.967578</td>\n",
       "      <td>P-0008152-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32052</th>\n",
       "      <td>3</td>\n",
       "      <td>2.633145</td>\n",
       "      <td>P-0032331-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32053</th>\n",
       "      <td>2</td>\n",
       "      <td>1.755430</td>\n",
       "      <td>P-0028037-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32054</th>\n",
       "      <td>11</td>\n",
       "      <td>10.821680</td>\n",
       "      <td>P-0009995-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32055</th>\n",
       "      <td>3</td>\n",
       "      <td>2.951367</td>\n",
       "      <td>P-0007699-T01-IM5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32056</th>\n",
       "      <td>5</td>\n",
       "      <td>4.388575</td>\n",
       "      <td>P-0021347-T01-IM6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32057</th>\n",
       "      <td>6</td>\n",
       "      <td>5.266290</td>\n",
       "      <td>P-0038539-T01-IM6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32058 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Nmut    Nmut_Mb Tumor_Sample_Barcode\n",
       "0         4   3.935156    P-0006360-T01-IM5\n",
       "1         1   0.877715    P-0029754-T01-IM6\n",
       "2         5   4.388575    P-0033656-T01-IM6\n",
       "3        10   8.777150    P-0022073-T01-IM6\n",
       "4        12  10.532580    P-0022194-T02-IM6\n",
       "5         6   5.902735    P-0005012-T01-IM5\n",
       "6         6   5.266290    P-0031527-T01-IM6\n",
       "7        20  17.554300    P-0030628-T01-IM6\n",
       "8         5   4.388575    P-0026122-T01-IM6\n",
       "9         6   5.266290    P-0015432-T01-IM6\n",
       "10        3   2.633145    P-0025446-T01-IM6\n",
       "11        9   8.854102    P-0009590-T01-IM5\n",
       "12        5   4.388575    P-0032598-T01-IM6\n",
       "13        1   1.115244    P-0002703-T01-IM3\n",
       "14        4   3.510860    P-0034494-T01-IM6\n",
       "15       18  15.798870    P-0032035-T01-IM6\n",
       "16        2   1.755430    P-0020236-T03-IM6\n",
       "17        2   1.755430    P-0015903-T01-IM6\n",
       "18        6   5.902735    P-0003514-T01-IM5\n",
       "19       16  14.043440    P-0018316-T01-IM6\n",
       "20       38  33.353170    P-0017068-T01-IM6\n",
       "21        3   2.633145    P-0024645-T01-IM6\n",
       "22        8   7.870313    P-0005271-T01-IM5\n",
       "23        6   5.902735    P-0004762-T01-IM5\n",
       "24        8   7.021720    P-0026370-T02-IM6\n",
       "25       11   9.654865    P-0017116-T01-IM6\n",
       "26        5   4.388575    P-0032590-T01-IM6\n",
       "27       12  10.532580    P-0036268-T01-IM6\n",
       "28       21  18.432015    P-0024571-T01-IM6\n",
       "29        6   5.902735    P-0009644-T01-IM5\n",
       "...     ...        ...                  ...\n",
       "32028     3   2.633145    P-0036060-T01-IM6\n",
       "32029     1   1.115244    P-0002203-T01-IM3\n",
       "32030     8   7.021720    P-0021281-T01-IM6\n",
       "32031     1   0.877715    P-0018310-T01-IM6\n",
       "32032     1   0.877715    P-0039611-T01-IM6\n",
       "32033     8   7.021720    P-0022819-T01-IM6\n",
       "32034     4   3.935156    P-0004807-T01-IM5\n",
       "32035     5   4.918946    P-0013870-T01-IM5\n",
       "32036     8   7.021720    P-0005143-T04-IM6\n",
       "32037    20  17.554300    P-0023469-T01-IM6\n",
       "32038     5   4.388575    P-0015602-T01-IM6\n",
       "32039     4   3.935156    P-0010234-T01-IM5\n",
       "32040     3   2.633145    P-0023987-T01-IM6\n",
       "32041     3   3.345731    P-0002254-T01-IM3\n",
       "32042     1   0.983789    P-0009059-T01-IM5\n",
       "32043     7   6.886524    P-0010267-T01-IM5\n",
       "32044     5   4.388575    P-0035066-T01-IM6\n",
       "32045     2   1.967578    P-0008052-T01-IM5\n",
       "32046     5   4.388575    P-0034412-T01-IM6\n",
       "32047     6   5.266290    P-0030111-T01-IM6\n",
       "32048     6   6.691462    P-0001376-T01-IM3\n",
       "32049     5   4.918946    P-0011240-T01-IM5\n",
       "32050     2   1.755430    P-0029163-T01-IM6\n",
       "32051     2   1.967578    P-0008152-T01-IM5\n",
       "32052     3   2.633145    P-0032331-T01-IM6\n",
       "32053     2   1.755430    P-0028037-T01-IM6\n",
       "32054    11  10.821680    P-0009995-T01-IM5\n",
       "32055     3   2.951367    P-0007699-T01-IM5\n",
       "32056     5   4.388575    P-0021347-T01-IM6\n",
       "32057     6   5.266290    P-0038539-T01-IM6\n",
       "\n",
       "[32058 rows x 3 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nmutDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "nmutDf.to_csv(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/nmutInfo_impact_filtered.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Get summarized observed oncogenic mutation data in a data frame**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "impactFilteredMaf = analysis_utils.load_in_df_with_progress(pathPrefix + '/ifs/work/taylorlab/friedman/myAdjustedDataFiles/tempScriptFiles/filteredMafAnnotated_trinuc.maf', 275171)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#create a unique identifier to use to calculate n distinct oncogenic mutations in a gene\n",
    "impactFilteredMaf['uuid'] = impactFilteredMaf.apply(lambda row: \n",
    "                                                    row['Hugo_Symbol'] + '_' + str(row['Start_Position']), axis=1)              \n",
    "impactFilteredMaf['pid'] = impactFilteredMaf['Tumor_Sample_Barcode'].apply(lambda x: x[:9])                              "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2818: DtypeWarning: Columns (5,14,18,24,25,26,28,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  if self.run_code(code, result):\n"
     ]
    }
   ],
   "source": [
    "cancerTypeDict = analysis_utils.get_cancer_type_information(cancerTypeDfPath = pathPrefix + '/ifs/work/taylorlab/friedman/mskImpactAsOfMarch2019/dmp/mskimpact/data_clinical_sample.txt', mode='pid')\n",
    "impactFilteredMaf['cancer_type'] = impactFilteredMaf['pid'].apply(lambda x: cancerTypeDict[x] if x in cancerTypeDict else None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Non-Small Cell Lung Cancer', 'Colorectal Cancer', 'Endometrial Cancer', 'Breast Cancer', 'Melanoma', 'Glioma', 'Bladder Cancer', 'Cancer of Unknown Primary', 'Pancreatic Cancer', 'Prostate Cancer', 'Colon Adenocarcinoma', 'Lung Adenocarcinoma', 'Uterine Endometrioid Carcinoma', 'Skin Cancer, Non-Melanoma', 'Esophagogastric Cancer', 'Hepatobiliary Cancer', 'Cutaneous Melanoma', 'Soft Tissue Sarcoma', 'Head and Neck Cancer', 'Ovarian Cancer', 'Small Cell Lung Cancer', 'Bladder Urothelial Carcinoma', 'Prostate Adenocarcinoma', 'Breast Invasive Ductal Carcinoma', 'Colorectal Adenocarcinoma', 'Renal Cell Carcinoma', 'Small Bowel Cancer', 'Glioblastoma Multiforme', 'Endometrial Carcinoma', 'Thyroid Cancer']\n"
     ]
    }
   ],
   "source": [
    "cancerTypesToAnalyze = [x[0] for x in Counter(impactFilteredMaf['cancer_type']).most_common(30)]\n",
    "print cancerTypesToAnalyze"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "impactGenes = set(['ABL1', 'ACVR1', 'AGO2', 'AKT1', 'AKT2', 'AKT3', 'ALK', 'ALOX12B', 'ANKRD11', 'APC', 'AR', 'ARAF', 'ARID1A', 'ARID1B', 'ARID2', 'ARID5B', 'ASXL1', 'ASXL2', 'ATM', 'ATR', 'ATRX', 'AURKA', 'AURKB', 'AXIN1', 'AXIN2', 'AXL', 'B2M', 'BABAM1', 'BAP1', 'BARD1', 'BBC3', 'BCL10', 'BCL2', 'BCL2L1', 'BCL2L11', 'BCL6', 'BCOR', 'BIRC3', 'BLM', 'BMPR1A', 'BRAF', 'BRCA1', 'BRCA2', 'BRD4', 'BRIP1', 'BTK', 'CALR', 'CARD11', 'CARM1', 'CASP8', 'CBFB', 'CBL', 'CCND1', 'CCND2', 'CCND3', 'CCNE1', 'CD274', 'CD276', 'CD79A', 'CD79B', 'CDC42', 'CDC73', 'CDH1', 'CDK12', 'CDK4', 'CDK6', 'CDK8', 'CDKN1A', 'CDKN1B', 'CDKN2A', 'CDKN2B', 'CDKN2C', 'CEBPA', 'CENPA', 'CHEK1', 'CHEK2', 'CIC', 'CREBBP', 'CRKL', 'CRLF2', 'CSDE1', 'CSF1R', 'CSF3R', 'CTCF', 'CTLA4', 'CTNNB1', 'CUL3', 'CXCR4', 'CYLD', 'CYSLTR2', 'DAXX', 'DCUN1D1', 'DDR2', 'DICER1', 'DIS3', 'DNAJB1', 'DNMT1', 'DNMT3A', 'DNMT3B', 'DOT1L', 'DROSHA', 'DUSP4', 'E2F3', 'EED', 'EGFL7', 'EGFR', 'EIF1AX', 'EIF4A2', 'EIF4E', 'ELF3', 'EP300', 'EPAS1', 'EPCAM', 'EPHA3', 'EPHA5', 'EPHA7', 'EPHB1', 'ERBB2', 'ERBB3', 'ERBB4', 'ERCC2', 'ERCC3', 'ERCC4', 'ERCC5', 'ERF', 'ERG', 'ERRFI1', 'ESR1', 'ETV1', 'ETV6', 'EZH1', 'EZH2', 'FAM123B', 'FAM175A', 'FAM46C', 'FAM58A', 'FANCA', 'FANCC', 'FAT1', 'FBXW7', 'FGF19', 'FGF3', 'FGF4', 'FGFR1', 'FGFR2', 'FGFR3', 'FGFR4', 'FH', 'FLCN', 'FLT1', 'FLT3', 'FLT4', 'FOXA1', 'FOXL2', 'FOXO1', 'FOXP1', 'FUBP1', 'FYN', 'GATA1', 'GATA2', 'GATA3', 'GLI1', 'GNA11', 'GNAQ', 'GNAS', 'GPS2', 'GREM1', 'GRIN2A', 'GSK3B', 'H3F3A', 'H3F3B', 'H3F3C', 'HGF', 'HIST1H1C', 'HIST1H2BD', 'HIST1H3A', 'HIST1H3B', 'HIST1H3C', 'HIST1H3D', 'HIST1H3E', 'HIST1H3F', 'HIST1H3G', 'HIST1H3H', 'HIST1H3I', 'HIST1H3J', 'HIST2H3C', 'HIST2H3D', 'HIST3H3', 'HLA-A', 'HLA-B', 'HNF1A', 'HOXB13', 'HRAS', 'ICOSLG', 'ID3', 'IDH1', 'IDH2', 'IFNGR1', 'IGF1', 'IGF1R', 'IGF2', 'IKBKE', 'IKZF1', 'IL10', 'IL7R', 'INHA', 'INHBA', 'INPP4A', 'INPP4B', 'INPPL1', 'INSR', 'IRF4', 'IRS1', 'IRS2', 'JAK1', 'JAK2', 'JAK3', 'JUN', 'KDM5A', 'KDM5C', 'KDM6A', 'KDR', 'KEAP1', 'KIT', 'KLF4', 'KMT2B', 'KMT5A', 'KNSTRN', 'KRAS', 'LATS1', 'LATS2', 'LMO1', 'LYN', 'MALT1', 'MAP2K1', 'MAP2K2', 'MAP2K4', 'MAP3K1', 'MAP3K13', 'MAP3K14', 'MAPK1', 'MAPK3', 'MAPKAP1', 'MAX', 'MCL1', 'MDC1', 'MDM2', 'MDM4', 'MED12', 'MEF2B', 'MEN1', 'MET', 'MGA', 'MITF', 'MLH1', 'KMT2A', 'KMT2B', 'KMT2C', 'MPL', 'MRE11A', 'MSH2', 'MSH3', 'MSH6', 'MSI1', 'MSI2', 'MST1', 'MST1R', 'MTOR', 'MUTYH', 'MYC', 'MYCL1', 'MYCN', 'MYD88', 'MYOD1', 'NBN', 'NCOA3', 'NCOR1', 'NEGR1', 'NF1', 'NF2', 'NFE2L2', 'NFKBIA', 'NKX2-1', 'NKX3-1', 'NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4', 'NPM1', 'NRAS', 'NSD1', 'NTHL1', 'NTRK1', 'NTRK2', 'NTRK3', 'NUF2', 'NUP93', 'PAK1', 'PAK7', 'PALB2', 'PARK2', 'PARP1', 'PAX5', 'PBRM1', 'PDCD1', 'PDCD1LG2', 'PDGFRA', 'PDGFRB', 'PDPK1', 'PGR', 'PHOX2B', 'PIK3C2G', 'PIK3C3', 'PIK3CA', 'PIK3CB', 'PIK3CD', 'PIK3CG', 'PIK3R1', 'PIK3R2', 'PIK3R3', 'PIM1', 'PLCG2', 'PLK2', 'PMAIP1', 'PMS1', 'PMS2', 'PNRC1', 'POLD1', 'POLE', 'PPARG', 'PPM1D', 'PPP2R1A', 'PPP4R2', 'PPP6C', 'PRDM1', 'PRDM14', 'PREX2', 'PRKAR1A', 'PRKCI', 'PRKD1', 'PTCH1', 'PTEN', 'PTP4A1', 'PTPN11', 'PTPRD', 'PTPRS', 'PTPRT', 'RAB35', 'RAC1', 'RAC2', 'RAD21', 'RAD50', 'RAD51', 'RAD51C', 'RAD51L1', 'RAD51L3', 'RAD52', 'RAD54L', 'RAF1', 'RARA', 'RASA1', 'RB1', 'RBM10', 'RECQL', 'RECQL4', 'REL', 'RET', 'RFWD2', 'RHEB', 'RHOA', 'RICTOR', 'RIT1', 'RNF43', 'ROS1', 'RPS6KA4', 'RPS6KB2', 'RPTOR', 'RRAGC', 'RRAS', 'RRAS2', 'RTEL1', 'RUNX1', 'RXRA', 'RYBP', 'SDHA', 'SDHAF2', 'SDHB', 'SDHC', 'SDHD', 'SESN1', 'SESN2', 'SESN3', 'SETD2', 'SF3B1', 'SH2B3', 'SH2D1A', 'SHOC2', 'SHQ1', 'SLX4', 'SMAD2', 'SMAD3', 'SMAD4', 'SMARCA4', 'SMARCB1', 'SMARCD1', 'SMO', 'SMYD3', 'SOCS1', 'SOS1', 'SOX17', 'SOX2', 'SOX9', 'SPEN', 'SPOP', 'SPRED1', 'SRC', 'SRSF2', 'STAG2', 'STAT3', 'STAT5A', 'STAT5B', 'STK11', 'STK19', 'STK40', 'SUFU', 'SUZ12', 'SYK', 'TAP1', 'TAP2', 'TBX3', 'TCEB1', 'TCF3', 'TCF7L2', 'TEK', 'TERT', 'TET1', 'TET2', 'TGFBR1', 'TGFBR2', 'TMEM127', 'TMPRSS2', 'TNFAIP3', 'TNFRSF14', 'TOP1', 'TP53', 'TP53BP1', 'TP63', 'TRAF2', 'TRAF7', 'TSC1', 'TSC2', 'TSHR', 'U2AF1', 'UPF1', 'VEGFA', 'VHL', 'VTCN1', 'WHSC1', 'WHSC1L1', 'WT1', 'WWTR1', 'XIAP', 'XPO1', 'XRCC2', 'YAP1', 'YES1', 'ZFHX3', 'ZRSR2'])\n",
    "listOfDicts = []\n",
    "for gene in impactGenes:\n",
    "    geneDf = impactFilteredMaf[impactFilteredMaf['Hugo_Symbol'] == gene]\n",
    "    oncogenicGeneDf = geneDf[geneDf['oncogenic'].notnull()]\n",
    "    d = {\n",
    "        'gene': gene,\n",
    "        'nDistinctMutations': len(set(geneDf['uuid'])),\n",
    "        'nMutations': geneDf.shape[0],\n",
    "        'nDistinctOncogenicMutations': len(set(oncogenicGeneDf['uuid'])),\n",
    "        'nOncogenicMutations': oncogenicGeneDf.shape[0]   \n",
    "    }\n",
    "    \n",
    "    #collect some data for cancerTypes (note we dont look at any old mutations at least for now)\n",
    "    for cancerType in cancerTypesToAnalyze:\n",
    "        ctGeneDf = oncogenicGeneDf[oncogenicGeneDf['cancer_type'] == cancerType]\n",
    "        d['nDistinctOncogenicMutations_' + cancerType] = len(set(ctGeneDf['uuid']))\n",
    "        d['nOncogenicMutations_' + cancerType] = ctGeneDf.shape[0]\n",
    "    \n",
    "    listOfDicts.append(d)\n",
    "dfMutationsInImpact = pd.DataFrame(listOfDicts)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "dfMutationsInImpact.to_csv(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/observedMutationCounts.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
