{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What we do here is load in the per gene file generated by the command **python compare_observed_and_expected_mut_burden.py casesFile mode** and reason on it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import sys\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "sys.path.append('/Users/friedman/Desktop/mnt/ifs/work/taylorlab/friedman/myUtils')\n",
    "import analysis_utils \n",
    "import mutationSigUtils \n",
    "import maf_analysis_utils\n",
    "import mutation_modeling_util"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathPrefix = '/Users/friedman/Desktop/mnt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'mutation_modeling_util' from '/Users/friedman/Desktop/mnt/ifs/work/taylorlab/friedman/myUtils/mutation_modeling_util.pyc'>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reload(mutation_modeling_util)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/simulatedMutationSummary.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "oncogenicityDict = mutation_modeling_util.calculate_quadnuc_based_oncogenic_susceptibility_dict(df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "considering the  25 th gene\n",
      "considering the  50 th gene\n",
      "considering the  75 th gene\n",
      "considering the  100 th gene\n",
      "considering the  125 th gene\n",
      "considering the  150 th gene\n",
      "considering the  175 th gene\n",
      "considering the  200 th gene\n",
      "considering the  225 th gene\n",
      "considering the  250 th gene\n",
      "considering the  275 th gene\n",
      "considering the  300 th gene\n",
      "considering the  325 th gene\n",
      "considering the  350 th gene\n",
      "considering the  375 th gene\n",
      "considering the  400 th gene\n",
      "considering the  425 th gene\n",
      "considering the  450 th gene\n"
     ]
    }
   ],
   "source": [
    "oncogenicityDictByGene = mutation_modeling_util.calculate_quadnuc_and_gene_based_oncogenic_susceptibility_dict(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2714: DtypeWarning: Columns (13,36,39,85,136) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "allImpactMuts = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/all_impact_mutations_annotated_cohort.maf')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Now enumerate observed vs expected data for genes based on their type and cancer type**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "we run the following on the cluster:\n",
    "python compare_observed_and_expected_mut_burden.py /ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/idFiles/endometrialHypermutantIds.txt perGene\n",
    "\n",
    "python compare_observed_and_expected_mut_burden.py /ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/idFiles/colorectalHypermutantIds.txt perGene\n",
    "\n",
    "python compare_observed_and_expected_mut_burden.py /ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/idFiles/endometrialHypermutantIds.txt perGene outfile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "endometrialDf = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/observedVsExpected/endometrialGenesObservedVsExpected.tsv')\n",
    "colorectalDf = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/observedVsExpected/colorectalGenesObservedVsExpected.tsv')\n",
    "gliomaDf = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/observedVsExpected/gliomaGenesObservedVsExpected.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Endometrial Cancer\n",
      "Glioma\n",
      "Colorectal Cancer\n",
      "Endometrial Cancer\n",
      "Glioma\n",
      "Colorectal Cancer\n"
     ]
    }
   ],
   "source": [
    "cancerTypeImplicationDictBig = maf_analysis_utils.create_dictionary_mapping_genes_to_cancer_types_with_implication(allImpactMuts, pathPrefix, cancerTypes=set(['Glioma', 'Colorectal Cancer', 'Endometrial Cancer']), t=.1)\n",
    "cancerTypeImplicationDictSmall = maf_analysis_utils.create_dictionary_mapping_genes_to_cancer_types_with_implication(allImpactMuts, pathPrefix, cancerTypes=set(['Glioma', 'Colorectal Cancer', 'Endometrial Cancer']), t=.025) \n",
    "for key, value in cancerTypeImplicationDictBig.items():\n",
    "    curCTypeGenes = cancerTypeImplicationDictSmall[key]\n",
    "    cancerTypeImplicationDictSmall[key] = curCTypeGenes - value\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "endometrialDf['cancer_type'] = 'Endometrial Cancer'\n",
    "colorectalDf['cancer_type'] = 'Colorectal Cancer'\n",
    "gliomaDf['cancer_type'] = 'Glioma'\n",
    "combinedDf = pd.concat([endometrialDf, colorectalDf, gliomaDf])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedDf['geneMutType'] = combinedDf.apply(lambda row:\n",
    "                                            'strongly_recurrent' if row['gene'] in cancerTypeImplicationDictBig[row['cancer_type']]\n",
    "                                             else 'weakly_recurrent' if row['gene'] in cancerTypeImplicationDictSmall[row['cancer_type']]\n",
    "                                            else 'not_recurrent', axis=1\n",
    "                                            )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedDf['nCases'] = combinedDf['cancer_type'].apply(lambda x: 53 if x == 'Glioma'\n",
    "                                                       else 261 if x == 'Colorectal Cancer' \n",
    "                                                       else 237)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedDf['geneObsExpectedRatio'] = combinedDf.apply(lambda row: 1.0*(row['nObservedMuts']- row['nExpectedMuts'])/row['nCases'], axis=1)\n",
    "combinedDf['geneDisplayName'] = combinedDf.apply(lambda row: \n",
    "                                                 row['gene'] if row['geneMutType'] == 'strongly_recurrent'\n",
    "                                                 else row['gene'] if row['geneObsExpectedRatio'] > .25\n",
    "                                                 else row['gene'] if row['geneObsExpectedRatio'] < -.1\n",
    "                                                 else None,axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedDf.to_csv('/Users/friedman/Desktop/WORK/dataForLocalPlotting/observedVsExpectedByGeneHypermutators.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Add hotspot info into data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cntr = 0\n",
    "d = pathPrefix + '/ifs/work/taylorlab/friedman/myAdjustedDataFiles/simulatedMafs/geneMutMafs/'\n",
    "listOfDicts = []\n",
    "for fPath in os.listdir(d):\n",
    "    if 'v2' in fPath:\n",
    "        print cntr\n",
    "        cntr+=1\n",
    "        \n",
    "        gene = fPath.split('_')[0]\n",
    "        df = pd.read_table(d + fPath)\n",
    "        hotspots = df[df['is-a-hotspot'].notnull()]\n",
    "        allPossibleQuadNucs = [firstNuc + change + lastNuc for firstNuc in ['A', 'T', 'C', 'G'] for change in ['CA', 'CG', 'CT', 'TA', 'TC', 'TG'] for lastNuc in ['A', 'T', 'C', 'G']]\n",
    "        nHotspots = hotspots.shape[0]\n",
    "        quadNucCounts = Counter(hotspots['quadNuc'])\n",
    "        \n",
    "        localD = {}\n",
    "        for quadNuc in allPossibleQuadNucs:\n",
    "            quadNucCount = 0\n",
    "            if quadNuc in quadNucCounts:\n",
    "                quadNucCount = quadNucCounts[quadNuc]\n",
    "            localD[quadNuc + '_hotspot'] = quadNucCount\n",
    "        localD['nHotspot'] = nHotspots\n",
    "        localD['Hugo_Symbol'] = gene\n",
    "        listOfDicts.append(localD)\n",
    "        \n",
    "hotspotDf = pd.DataFrame(listOfDicts)\n",
    "            \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_table(pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/simulatedMutationSummary.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "hotspotDfCopy = hotspotDf.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedDf = df.merge(hotspotDf, left_on='Hugo_Symbol', right_on='Hugo_Symbol')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedDf.to_csv('/Users/friedman/Desktop/workOffline/mutSimulationInfoIncludingHotspots.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**HOTSPOTS IN PANEL VS NOT ANALYSIS**\n",
    "<br/><br/><br/><br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#offline instead of imports\n",
    "\n",
    "import re\n",
    "\n",
    "def enumerate_recurrently_mutated_tumor_supressors_and_oncogenes(cohortMaf, thresh=.1):\n",
    "    tumorSupressors = set(['ERRFI1', 'ASXL2', 'PMAIP1', 'ACTG1', 'SUFU', 'FBXO11', 'MEN1', 'FAM58A', 'B2M', 'RB1', 'DUSP22', 'SESN1', 'GPS2', 'RAD51D', 'SMG1', 'CDC73', 'MAP3K1', 'SMARCB1', 'INPP4B', 'PARK2', 'SMAD4', 'CBFB', 'CDH1', 'PPP6C', 'SETDB1', 'SETDB2', 'NF2', 'CDKN2B', 'CDKN2C', 'CDKN2A', 'DDX3X', 'PIK3R1', 'BARD1', 'PDS5B', 'KLF4', 'SPRED1', 'VHL', 'SMAD2', 'PMS1', 'PMS2', 'SETD2', 'GATA3', 'TBL1XR1', 'MUTYH', 'SOCS1', 'FAM175A', 'ROBO1', 'ARID1B', 'ARID1A', 'TCF7L2', 'STK11', 'FOXA1', 'PTEN', 'FAT1', 'FAS', 'CYLD', 'MAX', 'SH2D1A', 'APC', 'NTHL1', 'CTCF', 'KDM5C', 'KMT2C', 'ZFHX3', 'FOXP1', 'PIGA', 'CDKN1B', 'CDKN1A', 'FUBP1', 'MSH2', 'ID3', 'TNFRSF14', 'TRAF3', 'EP400', 'BRIP1', 'ARID4A', 'ARID4B', 'XRCC2', 'DAXX', 'SDHAF2', 'ASXL1', 'AMER1', 'RASA1', 'EGR1', 'MST1', 'SOX17', 'RUNX1', 'PIK3R3', 'NCOR1', 'NF1', 'JAK1', 'PTPRD', 'CHEK2', 'CHEK1', 'SMC1A', 'TMEM127', 'STAG1', 'RAD51', 'TCF3', 'STAG2', 'ARID2', 'RAD50', 'RNF43', 'PARP1', 'BLM', 'CUX1', 'RECQL', 'RAD21', 'PTPN2', 'PTPN1', 'SLX4', 'INHA', 'PAX5', 'IRF1', 'TP53', 'HLA-A', 'IRF8', 'CBL', 'TOP1', 'SHQ1', 'PRDM1', 'NSD1', 'ATXN2', 'CREBBP', 'HDAC4', 'SESN2', 'PPP2R1A', 'EPHA7', 'ATM', 'EPHA3', 'POT1', 'SMAD3', 'MOB3B', 'TBX3', 'POLE', 'ATR', 'FANCD2', 'FH', 'BCORL1', 'SOX9', 'IKZF3', 'TSC1', 'TP63', 'MRE11A', 'SDHC', 'BTG1', 'POLD1', 'CIITA', 'SMC3', 'SAMHD1', 'RTEL1', 'ECT2L', 'PIK3R2', 'CRBN', 'FANCC', 'NBN', 'FANCA', 'HLA-B', 'RECQL4', 'DUSP4', 'ERCC2', 'FBXW7', 'TGFBR2', 'TGFBR1', 'MSH3', 'RBM15', 'TET1', 'TET3', 'SESN3', 'MGA', 'LTB', 'FOXL2', 'SH2B3', 'BCOR', 'HIST1H1D', 'ATRX', 'EP300', 'RAD51C', 'RAD51B', 'HIST1H1B', 'TNFAIP3', 'DICER1', 'ARID5B', 'LATS2', 'FOXO1', 'KEAP1', 'EZH2', 'SP140', 'NKX3-1', 'PBRM1', 'PALB2', 'CIC', 'BRCA1', 'DTX1', 'FLCN', 'SPEN', 'CD58', 'ERCC3', 'ERCC4', 'MSH6', 'BCL11B', 'BMPR1A', 'ERF', 'BRCA2', 'NOTCH2', 'EED', 'MITF', 'ELF3', 'SMARCA4', 'BBC3', 'ANKRD11', 'CEBPA', 'BCL2L11', 'AXIN2', 'AXIN1', 'CDK12', 'ESCO2', 'MLH1', 'SDHB', 'MED12', 'HNF1A', 'RYBP', 'ATP6V1B2', 'DNMT3B', 'KMT2B', 'KMT2A', 'DNMT3A', 'NFKBIA', 'TRAF5', 'KMT2D', 'SPOP', 'RBM10', 'P2RY8', 'TP53BP1', 'TSC2', 'KDM6A', 'EPCAM', 'PHOX2B', 'NPM1', 'BCL10', 'LATS1', 'HOXB13', 'ARID3A', 'PTPRT', 'PTPRS', 'INPPL1', 'NOTCH4', 'TET2', 'NOTCH1', 'CASP8', 'NOTCH3', 'GRIN2A', 'MAP2K4', 'WT1', 'BACH2', 'SDHA', 'BAP1', 'PTCH1', 'SDHD'])\n",
    "    occurenceCounter, rankingDict, fractionalDict = enumerate_top_n_oncogenic_mutated_genes_across_cohort(cohortMaf, n=50)\n",
    "    recurrentTumorSupressors = []\n",
    "    recurrentOncogenes = []\n",
    "    for key, value in fractionalDict.items():\n",
    "        if value > thresh:\n",
    "            if key in tumorSupressors:\n",
    "                recurrentTumorSupressors.append(key)\n",
    "            else:\n",
    "                recurrentOncogenes.append(key) \n",
    "    return set(recurrentTumorSupressors), set(recurrentOncogenes)\n",
    "\n",
    "def get_ids_by_hypermutant_status(hypermutantIdDir='/Users/friedman/Desktop/hypermutationStatusIdsOffline/', cancerType='', hypermutantStatus = 'Hypermutated'):\n",
    "    cancerTypeAdj = re.sub(' ', '_', cancerType)\n",
    "    path = os.path.join(hypermutantIdDir, cancerTypeAdj + '.tsv')\n",
    "    df = pd.read_table(path)\n",
    "    if hypermutantStatus == 'all':\n",
    "        return set(df['Tumor_Sample_Barcode'])\n",
    "    else:\n",
    "        return set(df[df['hypermutantClassification'] == hypermutantStatus]['Tumor_Sample_Barcode'])\n",
    "\n",
    "def create_reference_four_nuc(refTri, refAllele, altAllele, variantType):\n",
    "    #properly invert when needed\n",
    "    def invert_allele_and_ref_tri(altAllele):\n",
    "        nucleotideDict = {'A': 'T', 'G': 'C', 'C':'G', 'T':'A'}\n",
    "        return nucleotideDict[altAllele]\n",
    "\n",
    "    if variantType != 'SNP': return None  #there is no reference trinuc for non snps\n",
    "    if not isinstance(refTri, basestring): return None #if the ref tri is not a string we better return none\n",
    "    if len(refTri) < 3: return None # if the ref tri is less than length 3 (at the end of an exon), we cant do anything\n",
    "    refTri = str(refTri) #just make sure the ref tri is a string here to avoid funny business\n",
    "    refAlleleFromReftri = refTri[1]\n",
    "    alt = altAllele\n",
    "    if refAlleleFromReftri != refAllele:\n",
    "        alt = invert_allele_and_ref_tri(altAllele)\n",
    "    quadNuc = refTri[:2] + alt + refTri[2]\n",
    "    return quadNuc    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#FOR US TO DO OBSERVED VS EXPECTED\n",
    "def quantify_quadnuc_hotspot_susceptibility_per_mutation(simMafData, relatedGenes=None):\n",
    "    allBases = ['A', 'C', 'G', 'T']\n",
    "    changes = ['CA', 'CG', 'CT', 'TA', 'TC', 'TG'] #format: 'CA' means a change from C>A\n",
    "    allQuadNucs = [firstBase + change + lastBase for firstBase in allBases for change in changes for lastBase in allBases] #enumerate all 96 quadnucs for signatures\n",
    "    \n",
    "    #simulatedMutsRelated = simMafData[simMafData['Hugo_Symbol'].isin(relatedGenes)]\n",
    "    #simulatedMutsUnrelated = simMafData[~simMafData['Hugo_Symbol'].isin(relatedGenes)]\n",
    "    \n",
    "    d = {}\n",
    "    \n",
    "    for quadNuc in allQuadNucs:\n",
    "        nPossibleMuts = sum(simMafData[quadNuc]) - sum(simMafData[quadNuc + '_silent'])\n",
    "        nPossibleHotspotMuts = sum(simMafData[quadNuc + '_hotspot'])\n",
    "        #nPossibleMutsRelated = sum(simulatedMutsRelated[quadNuc]) - sum(simulatedMutsRelated[quadNuc + '_silent'])\n",
    "        #nPossibleMutsUnrelated = sum(simulatedMutsUnrelated[quadNuc]) - sum(simulatedMutsUnrelated[quadNuc + '_silent'])\n",
    "        #nPossibleHotspotMutRelated = sum(simulatedMutsRelated[quadNuc + '_hotspot'])\n",
    "        #nPossibleHotspotMutUnrelated = sum(simulatedMutsUnrelated[quadNuc + '_hotspot'])\n",
    "        #d[quadNuc + '_related_hotspot_rate'] = (1.0*nPossibleHotspotMutRelated)/(1.0*nPossibleMutsRelated)\n",
    "        #d[quadNuc + '_unrelated_hotspot_rate'] = (1.0*nPossibleHotspotMutUnrelated)/(1.0*nPossibleMutsUnrelated)\n",
    "        d[quadNuc + '_hotspot_rate'] = (1.0*nPossibleHotspotMuts)/(1.0*nPossibleMuts)\n",
    "        \n",
    "    return d\n",
    "\n",
    "def calculate_quadnuc_based_oncogenic_susceptibility_dict(simMafData):\n",
    "    d = {}\n",
    "    allBases = ['A', 'C', 'G', 'T']\n",
    "    changes = ['CA', 'CG', 'CT', 'TA', 'TC', 'TG'] #format: 'CA' means a change from C>A\n",
    "    allQuadNucs = [firstBase + change + lastBase for firstBase in allBases for change in changes for lastBase in allBases] #enumerate all 96 quadnucs for signatures\n",
    "    for quadNuc in allQuadNucs:\n",
    "        nPossibleMuts = sum(simMafData[quadNuc]) - sum(simMafData[quadNuc + '_silent'])\n",
    "        nPossibleOncogenicMuts = sum(simMafData[quadNuc + '_oncogenic'])\n",
    "        d[quadNuc + '_oncogenic_rate'] = (1.0*nPossibleOncogenicMuts)/(1.0*nPossibleMuts)\n",
    "    return d\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_observed_and_expected_by_mut_type_hotspot(allMuts, ids, relatedGenes, hotspotSusceptibilityD):\n",
    "    listOfDicts = []\n",
    "    for case in set(ids):\n",
    "        caseMuts = allMuts[allMuts['Tumor_Sample_Barcode'] == case]\n",
    "        quadNucCounts = Counter(caseMuts['quadNuc'])\n",
    "        nmut = caseMuts.shape[0]\n",
    "\n",
    "        caseHotspotMuts = caseMuts[caseMuts['is-a-hotspot'] == 'Y']\n",
    "\n",
    "        nHotspotMuts = caseHotspotMuts.shape[0]\n",
    "        #nCaseRelatedHotspotMuts = caseHotspotMuts[caseHotspotMuts['Hugo_Symbol'].isin(relatedGenes)].shape[0]\n",
    "        #nCaseUnrelatedHotspotMuts = caseHotspotMuts[~caseHotspotMuts['Hugo_Symbol'].isin(relatedGenes)].shape[0]\n",
    "\n",
    "        #nExpectedHotspotsRelated = 0\n",
    "        #nExpectedHotspotsUnrelated = 0\n",
    "        nExpectedHotspots = 0\n",
    "        for quadNuc, count in quadNucCounts.items():\n",
    "            if quadNuc != None:\n",
    "                nExpectedHotspots += count*hotspotSusceptibilityD[quadNuc + '_hotspot_rate']\n",
    "                #nExpectedHotspotsRelated += count*hotspotSusceptibilityD[quadNuc + '_related_hotspot_rate']\n",
    "                #nExpectedHotspotsUnrelated += count*hotspotSusceptibilityD[quadNuc + '_unrelated_hotspot_rate']\n",
    "\n",
    "        listOfDicts.append({'Tumor_Sample_Barcode': case, \n",
    "                            #'nRelatedHotspotsObserved': nCaseRelatedHotspotMuts, 'nUnrelatedHotspotsObserved': nCaseUnrelatedHotspotMuts,\n",
    "                            #'nHotspotsObserved': nCaseRelatedHotspotMuts + nCaseUnrelatedHotspotMuts,\n",
    "                           #'nExpectedHotspotRelated': nExpectedHotspotsRelated, 'nExpectedHotspotsUnrelated':  nExpectedHotspotsUnrelated,\n",
    "                            #'nExpectedHotspots': nExpectedHotspotsRelated + nExpectedHotspotsUnrelated,\n",
    "                           'nHotspotObserved': nHotspotMuts, 'nHotspotExpected': nExpectedHotspots,\n",
    "                            'Nmut': nmut})\n",
    "\n",
    "    obsVsExpected = pd.DataFrame(listOfDicts)\n",
    "    return obsVsExpected\n",
    "    \n",
    "    \n",
    "\n",
    "def calculate_observed_and_expected_by_mut_type_oncogenic_version(allMuts, ids, relatedGenes, oncogenicSusceptibilityDict):\n",
    "    listOfDicts = []\n",
    "    cntr = 0\n",
    "    for case in set(ids):\n",
    "        \n",
    "        if cntr %100 == 0: print cntr\n",
    "        cntr += 1\n",
    "        \n",
    "        caseMuts = allMuts[(allMuts['Tumor_Sample_Barcode'] == case) & (allMuts['Variant_Type'] == 'SNP')]\n",
    "        quadNucCounts = Counter(caseMuts['quadNuc'])\n",
    "        nmut = caseMuts.shape[0]\n",
    "\n",
    "        caseOncogenicMuts = caseMuts[caseMuts['oncogenic'].notnull()]\n",
    "        nCaseOncogenicMuts = caseOncogenicMuts.shape[0]\n",
    "        \n",
    "        nExpectedOncogenic = 0\n",
    "        for quadNuc, count in quadNucCounts.items():\n",
    "            if quadNuc != None:\n",
    "                nExpectedOncogenic += count*oncogenicSusceptibilityDict[quadNuc + '_oncogenic_rate']\n",
    "        listOfDicts.append({'Tumor_Sample_Barcode': case, \n",
    "                            'nOncogenicObserved': nCaseOncogenicMuts,\n",
    "                            'nExpectedOncogenic': nExpectedOncogenic,\n",
    "                           'Nmut': nmut})\n",
    "\n",
    "    obsVsExpected = pd.DataFrame(listOfDicts)\n",
    "    return obsVsExpected\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2714: DtypeWarning: Columns (13,36,39,85,136) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "allImpactMuts = pd.read_table('/Users/friedman/Desktop/workOffline/all_impact_mutations_annotated_cohort.maf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#do it with this later\n",
    "#enumerate_recurrently_mutated_tumor_supressors_and_oncogenes(cohortMaf, thresh=.1)\n",
    "endometrialCancerGenes = set(['POLE', 'MLH1', 'MSH2', 'MSH6', 'PMS2','PTEN', 'BCOR', 'PIK3CA', 'ARID1A', 'TP53', 'PIK3R1', 'FBXW7', 'KRAS', 'CTNNB1', 'PPP2R1A', 'FGFR'])\n",
    "colorectalCancerGenes = set(['APC', 'PIK3CA', 'TP53', 'KRAS', 'SMAD4', 'ARID1A', 'PTEN', 'FBXW7', 'TCF7L2', 'SOX9', 'BRAF', 'POLE', 'MLH1', 'MSH2', 'MSH6', 'PMS2'])\n",
    "gliomaGenes = set(['TP53', 'TERT', 'PIK3CA', 'NF1', 'IDH1', 'PTEN', 'ATRX', 'EGFR', 'CIC', 'BRAF', 'KRAS'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "endometrialIds = analysis_utils.get_ids_by_hypermutant_status(hypermutantIdDir=pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds', cancerType='Endometrial Cancer', hypermutantStatus = 'Hypermutated')\n",
    "colorectalIds = analysis_utils.get_ids_by_hypermutant_status(hypermutantIdDir=pathPrefix +'/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds', cancerType='Colorectal Cancer', hypermutantStatus = 'Hypermutated')\n",
    "gliomaIds = analysis_utils.get_ids_by_hypermutant_status(hypermutantIdDir=pathPrefix +'/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds', cancerType='Glioma', hypermutantStatus = 'Hypermutated')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "endometrialCancerHyperIds = get_ids_by_hypermutant_status(cancerType='Endometrial Cancer', hypermutantStatus = 'Hypermutated')\n",
    "colorectalCancerHyperIds = get_ids_by_hypermutant_status(cancerType='Colorectal Cancer', hypermutantStatus = 'Hypermutated')\n",
    "gliomaCancerHyperIds = get_ids_by_hypermutant_status(cancerType='Glioma', hypermutantStatus = 'Hypermutated')\n",
    "\n",
    "endometrialCancerIds = get_ids_by_hypermutant_status(cancerType='Endometrial Cancer', hypermutantStatus = 'all')\n",
    "colorectalCancerIds = get_ids_by_hypermutant_status(cancerType='Colorectal Cancer', hypermutantStatus = 'all')\n",
    "gliomaCancerIds = get_ids_by_hypermutant_status(cancerType='Glioma', hypermutantStatus = 'all')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "allImpactMuts['quadNuc'] = allImpactMuts.apply(lambda row: mutationSigUtils.create_reference_four_nuc(row['Ref_Tri'], row['Reference_Allele'], row['Tumor_Seq_Allele2'], row['Variant_Type']), axis=1)\n",
    "#endometrialCancerHyperMuts = allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(endometrialCancerHyperIds)]\n",
    "#endometrialCancerHotspots = endometrialCancerHyperMuts[endometrialCancerHyperMuts['is-a-hotspot'] == 'Y']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "simulatedDataSummary = pd.read_table('/Users/friedman/Desktop/workOffline/mutSimulationInfoIncludingHotspots.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "hotspotSusceptibilityD = quantify_quadnuc_hotspot_susceptibility_per_mutation(simulatedDataSummary)\n",
    "oncogenicityDict = calculate_quadnuc_based_oncogenic_susceptibility_dict(simulatedDataSummary)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [],
   "source": [
    "colorectalDf = calculate_observed_and_expected_by_mut_type_hotspot(allImpactMuts, colorectalCancerIds, colorectalCancerGenes, hotspotSusceptibilityD)\n",
    "endometrialDf = calculate_observed_and_expected_by_mut_type_hotspot(allImpactMuts, endometrialCancerIds, endometrialCancerGenes, hotspotSusceptibilityD)\n",
    "gliomaDf = calculate_observed_and_expected_by_mut_type_hotspot(allImpactMuts, gliomaCancerIds, gliomaGenes, hotspotSusceptibilityD)\n",
    "\n",
    "#colorectalDf = calculate_observed_and_expected_by_mut_type_oncogenic_version(allImpactMuts, colorectalCancerIds, colorectalCancerGenes, oncogenicityDict)\n",
    "#endometrialDf = calculate_observed_and_expected_by_mut_type_oncogenic_version(allImpactMuts, endometrialCancerIds, endometrialCancerGenes, oncogenicityDict)\n",
    "#gliomaDf = calculate_observed_and_expected_by_mut_type_oncogenic_version(allImpactMuts, gliomaCancerIds, gliomaGenes, oncogenicityDict)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [],
   "source": [
    "colorectalDf['cancer_type'] = 'Colorectal Cancer'\n",
    "endometrialDf['cancer_type'] = 'Endometrial Cancer'\n",
    "gliomaDf['cancer_type'] = 'Glioma'\n",
    "\n",
    "combinedDf = pd.concat([colorectalDf, endometrialDf, gliomaDf])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedDf.to_csv('/Users/friedman/Desktop/WORK/dataForLocalPlotting/obsVsExpectedHotspots.tsv', index=False, sep='\\t')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**THIS SECTION IS CURRENTLY FOR US JUST TO SHOW THE DIFFERENT HOTSPOT SUSCEPTIBILITIES AND OBSERVEDS**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_n_hotspots_or_oncogenic_per_case(cohortMaf, label, mode='hotspot'):\n",
    "    \n",
    "    mafOfInterest = None\n",
    "    \n",
    "    if mode == 'hotspot':\n",
    "        mafOfInterest = cohortMaf[cohortMaf['is-a-hotspot'] == 'Y']\n",
    "    elif mode == 'oncogenic':\n",
    "        mafOfInterest = cohortMaf[(cohortMaf['oncogenic'].notnull()) & (cohortMaf['Variant_Type'] == 'SNP')]\n",
    "\n",
    "        \n",
    "    countsD = dict(mafOfInterest['Tumor_Sample_Barcode'].value_counts())\n",
    "    for case in set(cohortMaf['Tumor_Sample_Barcode']):\n",
    "        if case not in countsD:\n",
    "            countsD[case] = 0\n",
    "    \n",
    "    nmutD = dict(cohortMaf['Tumor_Sample_Barcode'].value_counts())\n",
    "    for case in set(cohortMaf['Tumor_Sample_Barcode']):\n",
    "        if case not in nmutD:\n",
    "            nmutD[case] = 0\n",
    "            \n",
    "    listOfDicts = []\n",
    "    for key, values in nmutD.items():\n",
    "        nmut = nmutD[key]\n",
    "        nHotspotOrOnc = countsD[key]\n",
    "        hotspotOrOncFrac = 0\n",
    "        if nmut != 0:\n",
    "            hotspotOrOncFrac = (1.0*nHotspotOrOnc)/nmut\n",
    "        \n",
    "        localD = {'id': key, 'class': label}\n",
    "        if mode == 'hotspot':\n",
    "            localD['hotspotFrac'] = hotspotOrOncFrac\n",
    "        elif mode == 'oncogenic':\n",
    "            localD['oncogenicFrac'] = hotspotOrOncFrac\n",
    "        \n",
    "        listOfDicts.append(localD)\n",
    "        \n",
    "    return pd.DataFrame(listOfDicts)\n",
    "\n",
    "\n",
    "#converts the dictionary of hypermutation\n",
    "def convert_hotspot_or_oncogenic_dict_to_df_format(d, mode='hotspot'):\n",
    "    listOfDicts = []\n",
    "    for key, value in d.items():\n",
    "        localD = {\n",
    "            'class': 'theoreticalSusceptibility',\n",
    "            'id': key\n",
    "        }\n",
    "        if mode == 'hotspot':\n",
    "            localD['hotspotFrac'] = value\n",
    "        elif mode == 'oncogenic':\n",
    "            localD['oncogenicFrac'] = value\n",
    "        listOfDicts.append(localD)\n",
    "    return pd.DataFrame(listOfDicts)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "endometrialHyperIds = analysis_utils.get_ids_by_hypermutant_status(cancerType='Endometrial Cancer', hypermutantStatus = 'Hypermutated', hypermutantIdDir= pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds')\n",
    "colorectalHyperIds = analysis_utils.get_ids_by_hypermutant_status(cancerType='Colorectal Cancer', hypermutantStatus = 'Hypermutated', hypermutantIdDir= pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds')\n",
    "gliomaHyperIds = analysis_utils.get_ids_by_hypermutant_status(cancerType='Glioma', hypermutantStatus = 'Hypermutated', hypermutantIdDir= pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds')\n",
    "\n",
    "endometrialNormalIds = analysis_utils.get_ids_by_hypermutant_status(cancerType='Endometrial Cancer', hypermutantStatus = 'Normal', hypermutantIdDir= pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds')\n",
    "colorectalNormalIds = analysis_utils.get_ids_by_hypermutant_status(cancerType='Colorectal Cancer', hypermutantStatus = 'Normal', hypermutantIdDir= pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds')\n",
    "gliomaNormalIds = analysis_utils.get_ids_by_hypermutant_status(cancerType='Glioma', hypermutantStatus = 'Normal', hypermutantIdDir= pathPrefix + '/ifs/work/taylorlab/friedman/hypermutationAnalysisProj/projectDataAndConfigFiles/hypermutationStatusIds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = 'oncogenic'\n",
    "#m = 'hotspot'\n",
    "endoNormalDf = count_n_hotspots_or_oncogenic_per_case(allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(endometrialNormalIds)], 'endometrialNormal', mode=m)\n",
    "colorectalNormalDf = count_n_hotspots_or_oncogenic_per_case(allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(colorectalNormalIds)], 'colorectalNormal', mode=m)\n",
    "gliomaNormalDf = count_n_hotspots_or_oncogenic_per_case(allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(gliomaNormalIds)], 'gliomaNormal', mode=m)\n",
    "\n",
    "endoHyperDf = count_n_hotspots_or_oncogenic_per_case(allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(endometrialHyperIds)], 'endometrialHyper', mode=m)\n",
    "colorectalHyperDf = count_n_hotspots_or_oncogenic_per_case(allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(colorectalHyperIds)], 'colorectalHyper', mode=m)\n",
    "gliomaHyperDf = count_n_hotspots_or_oncogenic_per_case(allImpactMuts[allImpactMuts['Tumor_Sample_Barcode'].isin(gliomaHyperIds)], 'gliomaHyper', mode=m)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "hypotheticalDfHotspots = convert_hotspot_or_oncogenic_dict_to_df_format(hotspotSusceptibilityD, mode = 'hotspot')\n",
    "hypotheticalDfOncogenic = convert_hotspot_or_oncogenic_dict_to_df_format(oncogenicityDict, mode = 'oncogenic')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:2: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "combinedDf = pd.concat([endoNormalDf, colorectalNormalDf, gliomaNormalDf,\n",
    "                       endoHyperDf, colorectalHyperDf, gliomaHyperDf, hypotheticalDfHotspots, hypotheticalDfOncogenic])\n",
    "combinedDf['simplifiedClass'] = combinedDf['class'].apply(lambda x: 'Hyper' if 'Hyper' in x \n",
    "                                                         else 'Normal' if 'Normal' in x else x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "#combinedDf.to_csv('/Users/friedman/Desktop/WORK/hotspotFractionOfMutationBurden.tsv', index=False, sep='\\t')\n",
    "combinedDf.to_csv('/Users/friedman/Desktop/WORK/oncogenicFractionOfMutationBurden.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.00417333686624\n"
     ]
    }
   ],
   "source": [
    "#max(combinedDf[combinedDf['class'] == 'theoreticalSusceptibility']['hotspotFrac'])\n",
    "#combinedDf[combinedDf['class'] == 'theoreticalSusceptibility']\n",
    "\n",
    "\n",
    "print max(hypotheticalDfHotspotsSigs['hotspotFrac'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Give information about mutation signatures and susceptibility**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert_quadnuc_df_to_mut_sigs_df(quadNucDf, colName):\n",
    "    quadNucDict = dict(zip(quadNucDf['id'].apply(lambda x: x[:4]), quadNucDf[colName]))\n",
    "    spectraD = mutationSigUtils.convert_spectrum_file_to_dict_of_dicts(spectrumFile=pathPrefix + '/ifs/work/taylorlab/friedman/noahFirstProject/signature_sig_copy/mutation-signatures/Stratton_signatures30.txt')\n",
    "    d = {}\n",
    "    for signature in spectraD.keys():\n",
    "        s = 0\n",
    "        for quadNuc, frac in spectraD[signature].items():\n",
    "            s += frac*quadNucDict[quadNuc]\n",
    "        d[signature] = s\n",
    "    return d\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "dHotspot = convert_quadnuc_df_to_mut_sigs_df(hypotheticalDfHotspots, 'hotspotFrac')\n",
    "dOncogenic = convert_quadnuc_df_to_mut_sigs_df(hypotheticalDfOncogenic, 'oncogenicFrac')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "hypotheticalDfHotspotsSigs = convert_hotspot_or_oncogenic_dict_to_df_format(dHotspot, mode = 'hotspot')\n",
    "hypotheticalDfOncogenicSigs = convert_hotspot_or_oncogenic_dict_to_df_format(dOncogenic, mode = 'oncogenic')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "#combinedDf = pd.concat([endoNormalDf, colorectalNormalDf, gliomaNormalDf,\n",
    "#                       endoHyperDf, colorectalHyperDf, gliomaHyperDf, hypotheticalDfHotspotsSigs])\n",
    "combinedDf = pd.concat([endoNormalDf, colorectalNormalDf, gliomaNormalDf,\n",
    "                       endoHyperDf, colorectalHyperDf, gliomaHyperDf, hypotheticalDfOncogenicSigs])\n",
    "\n",
    "\n",
    "combinedDf['simplifiedClass'] = combinedDf['class'].apply(lambda x: 'Hyper' if 'Hyper' in x \n",
    "                                                         else 'Normal' if 'Normal' in x else x)\n",
    "combinedDf.to_csv('/Users/friedman/Desktop/WORK/hotspotFractionOfMutationBurden.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Info for hotspot frac amongst all observed mutations only** <br/> <br/>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "allImpactMuts['variantUUID'] = allImpactMuts.apply(lambda row: str(row['Start_Position']) + '_' + str(row['Tumor_Seq_Allele2']), axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "allImpactMutsUnique = allImpactMuts.drop_duplicates(subset=['variantUUID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.2030494302811706"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "max(hypotheticalDfOncogenic[hypotheticalDfOncogenic['class'] == 'theoreticalSusceptibility']['oncogenicFrac'])\n",
    "#hypotheticalDfOncogenic\n",
    "\n",
    "\n",
    "#print hypotheticalDfOncogenic[(hypotheticalDfOncogenic['class'] == 'theoreticalSusceptibility') & (hypotheticalDfOncogenic['oncogenicFrac'] > .2)]\n",
    "\n",
    "#hypotheticalDfOncogenicSigs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GTAC 0.135746606335\n",
      "GTAA 0.258741258741\n",
      "GTAG 0.145488029466\n",
      "GTAT 0.0812641083521\n",
      "TTAG 0.518438177874\n",
      "TTAC 0.309771309771\n",
      "TTAA 0.57650273224\n",
      "TTAT 0.176356589147\n",
      "GCTG 0.0684299554052\n",
      "GCTC 0.0440528634361\n",
      "GCTA 0.27494199536\n",
      "ATAA 0.145679012346\n",
      "ATAC 0.119225037258\n",
      "CCAT 0.137020566982\n",
      "ATAG 0.136212624585\n",
      "GCTT 0.0504396112911\n",
      "CCAA 0.121383647799\n",
      "CCAC 0.0744732974032\n",
      "ATAT 0.112012987013\n",
      "CCAG 0.0955357142857\n",
      "TCTT 0.0682471264368\n",
      "ACTT 0.0955823293173\n",
      "TCTG 0.0890282131661\n",
      "CTCA 0.251644736842\n",
      "CTCC 0.0518518518519\n",
      "TCTC 0.0432801822323\n",
      "TCTA 0.190149103702\n",
      "ACTG 0.0807099091485\n",
      "CTAG 0.208181818182\n",
      "CTAC 0.092084006462\n",
      "CTAA 0.392971246006\n",
      "GTGA 0.404958677686\n",
      "GTGC 0.0909090909091\n",
      "GTGG 0.149877149877\n",
      "ACGA 0.158974358974\n",
      "ACGC 0.256916996047\n",
      "ACGG 0.177130044843\n",
      "GTGT 0.0593220338983\n",
      "TTCT 0.0435540069686\n",
      "TTCG 0.0586552217454\n",
      "TCAC 0.29542042042\n",
      "TTCA 0.0742268041237\n",
      "TTCC 0.0524861878453\n",
      "ATCT 0.0459126539754\n",
      "CCGG 0.111545988258\n",
      "CTGC 0.0598086124402\n",
      "CCGC 0.0675944333996\n",
      "CCGA 0.0914634146341\n",
      "ATCG 0.0765550239234\n",
      "CCTT 0.118991596639\n",
      "ATCA 0.0586932447398\n",
      "CCTC 0.0483263598326\n",
      "CCGT 0.141316073355\n",
      "CCTG 0.0905912269549\n",
      "ACTC 0.205450733753\n",
      "TCGC 0.0387643852211\n",
      "TCGA 0.229465449804\n",
      "TCGG 0.0732394366197\n",
      "CTCG 0.137614678899\n",
      "TCGT 0.0638497652582\n",
      "ATCC 0.0687203791469\n",
      "ACAG 0.140350877193\n",
      "ACAC 0.239273927393\n",
      "ACAA 0.115226337449\n",
      "ACAT 0.126465661642\n",
      "ACGT 0.150318471338\n",
      "CCTA 0.31830008673\n",
      "CTCT 0.072864321608\n",
      "GCAG 0.0722748815166\n",
      "GCAA 0.0647540983607\n",
      "GCAC 0.0687347931873\n",
      "GCAT 0.0564042303173\n",
      "CTGT 0.0550964187328\n",
      "TCAT 0.218513853904\n",
      "CTAT 0.120956399437\n",
      "TCAA 0.464806866953\n",
      "CTGG 0.165625\n",
      "CTGA 0.360544217687\n",
      "TCAG 0.386973180077\n",
      "GTCG 0.0989096573209\n",
      "GTCA 0.185003503854\n",
      "GTCC 0.0566727605119\n",
      "GTCT 0.042471042471\n",
      "TTGT 0.0519801980198\n",
      "TTGA 0.41958041958\n",
      "TTGC 0.0810810810811\n",
      "TTGG 0.111420612813\n",
      "ATGC 0.16077170418\n",
      "ATGA 0.189573459716\n",
      "ATGG 0.201834862385\n",
      "ACTA 0.258568851473\n",
      "GCGC 0.0575\n",
      "GCGA 0.0663265306122\n",
      "GCGG 0.0816831683168\n",
      "ATGT 0.103646833013\n",
      "GCGT 0.0623100303951\n",
      "None"
     ]
    },
    {
     "ename": "ZeroDivisionError",
     "evalue": "float division by zero",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mZeroDivisionError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-108-1dd51645696b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mquadNuc\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mallImpactMutsUnique\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'quadNuc'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m     \u001b[0mquadNucMuts\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mallImpactMutsUnique\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mallImpactMutsUnique\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'quadNuc'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mquadNuc\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0;32mprint\u001b[0m \u001b[0mquadNuc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;36m1.0\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mquadNucMuts\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mquadNucMuts\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'oncogenic'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnotnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0mquadNucMuts\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mZeroDivisionError\u001b[0m: float division by zero"
     ]
    }
   ],
   "source": [
    "for quadNuc in set(allImpactMutsUnique['quadNuc']):\n",
    "    quadNucMuts = allImpactMutsUnique[allImpactMutsUnique['quadNuc'] == quadNuc]\n",
    "    print quadNuc, (1.0*quadNucMuts[quadNucMuts['oncogenic'].notnull()].shape[0])/quadNucMuts.shape[0]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
